core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -376,12 +381,36 @@ class Partition(val topic: String,
   }
 
   /**
+   * Check and maybe increment the low watermark of the partition;
+   * this function can be triggered when any replica's logStartOffset changed
```

Comment:
It seems OK because when there is no follower, LW is only used in the `PurgeResponse` to tell user whether purge has finished. In this case the `PurgeRequest` would have triggered `maybeIncrementLeaderLW`.